consensus/src/quorum_store/batch_store.rs

```diff
@@ -308,7 +308,17 @@ impl<T: QuorumStoreSender + Clone + Send + Sync + 'static> BatchStore<T> {
         ret
     }
 
-    pub fn persist(&self, persist_request: PersistedValue) -> Option<SignedBatchInfo> {
+    pub fn persist(&self, persist_requests: Vec<PersistedValue>) -> Vec<SignedBatchInfo> {
+        let mut signed_infos = vec![];
+        for persist_request in persist_requests.into_iter() {
+            if let Some(signed_info) = self.persist_inner(persist_request) {
```

Comment:
do we need to handle partial failures now?

Reply:
I think the current behavior, that allows to fail some and succeed others in the same message, should be fine. Anyway what will matter if the batch reached quorum or not.